### PR TITLE
Global CurveManager

### PR DESCRIPTION
**Pull request by mitchute**
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Myoldmopar** [comment]
CI messages are not related to this branch, looks like it is generally happy here.

**Comment by mitchute** [comment]
@Myoldmopar I'll hold short of pushing this all the way through until we have the globals redesign in place. Feel free to merge, or we can hold off and finish it when that's done. Either is fine with me.

**Comment by mitchute** [comment]
I'm done with this if CI is happy.

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc [hunk 1]
```diff
@@ -1405,7 +1405,7 @@ namespace CoolingPanelSimple {
                 // Now, distribute the radiant energy of all systems to the appropriate surfaces, to people, and the air
                 DistributeCoolingPanelRadGains(state.dataChilledCeilingPanelSimple);
                 // Now "simulate" the system by recalculating the heat balances
-                HeatBalanceSurfaceManager::CalcHeatBalanceOutsideSurf(state.dataConvectionCoefficients, state.files, ZoneNum);
+                HeatBalanceSurfaceManager::CalcHeatBalanceOutsideSurf(state, state.dataConvectionCoefficients, state.files, ZoneNum);
```
Comment (mjwitte) [inline src/EnergyPlus/ChilledCeilingPanelSimple.cc:1408]: @mitchute Just skimming the changes and happened to see 3 different `state` arguments here. Is that intended?

File: src/EnergyPlus/EMSManager.cc [hunk 2]
```diff
@@ -302,7 +302,7 @@ namespace EMSManager {
             PluginManagement::onBeginEnvironment();
         }
 
-        InitEMS(state.files, iCalledFrom);
+        InitEMS(state, state.files, iCalledFrom);
```
Comment (mjwitte) [inline src/EnergyPlus/EMSManager.cc:305]: And here.

File: src/EnergyPlus/EMSManager.cc [hunk 3]
```diff
@@ -325,15 +325,15 @@ namespace EMSManager {
...
+                        EvaluateStack(state, state.files, EMSProgramCallManager(ProgramManagerNum).ErlProgramARR(ErlProgramNum));
                         anyProgramRan = true;
                     }
                 }
             }
         } else { // call specific program manager
             if (present(ProgramManagerToRun)) {
                 for (ErlProgramNum = 1; ErlProgramNum <= EMSProgramCallManager(ProgramManagerToRun).NumErlPrograms; ++ErlProgramNum) {
-                    EvaluateStack(state.files, EMSProgramCallManager(ProgramManagerToRun).ErlProgramARR(ErlProgramNum));
+                    EvaluateStack(state, state.files, EMSProgramCallManager(ProgramManagerToRun).ErlProgramARR(ErlProgramNum));
```
Comment (mjwitte) [inline src/EnergyPlus/EMSManager.cc:336]: And here.

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc (hunk 1 above)
Comment (mitchute) [reply src/EnergyPlus/ChilledCeilingPanelSimple.cc:1408]: @mjwitte The DataAirLoop PR (#8210) will clean up a bunch of these. That will also be followed by a couple of other subsequent PRs to just pass ```state``` in everywhere. We were originally trying to keep it as low-level as possible, but it's been causing a merge nightmare for @brianlball, @Myoldmopar, and I. Once this is all done, we may make another pass where we move to only pass the pieces of ```state``` that are actually needed. For now, though, we're just gonna pass it to minimize conflicts.

**Comment by Myoldmopar** [comment]
Pulled develop in and ran unit tests, everything looks good.  I anticipate this dropping in here in a few minutes.  I understand this will cause some conflicts, but I'd need to pull develop into any of the other branches I review this afternoon anyway, and the conflicts will be straightforward with these changes.

**Comment by Myoldmopar** [comment]
Good first result.  If they all look good I'll merge it.

**Comment by Myoldmopar** [comment]
OK, so I'm not sure how I accidentally duplicated a line in the license text of a file there, maybe I `ctrl-d`'d while thinking I had a different window focused.  Anywho, I verified the issue locally by running license-check, and then I fixed that and it passes.  I'm going to push that up and merge to avoid waiting on another round of CI.  Thanks @mitchute !

**Pull request merged**